package one.clproto.remote_file_management;

message RemoteFileMangement {
    required string message_type = 1;
    required bytes input = 2;
    optional string space_id = 3 [default = ""];
}

message GetAttr {
    required string file_id = 1;
}

message StorageAttibutes {
    optional string answer = 1 [default = "ok"];
    optional int64 dev = 2 [default = 0];
    optional int64 ino = 3 [default = 0];
    optional int64 mode = 4 [default = 0];
    optional int64 nlink = 5 [default = 1];
    optional int64 uid = 6 [default = 0];
    optional int64 gid = 7 [default = 0];
    optional int64 rdev = 8 [default = 0];
    optional int64 size = 9 [default = 0];
    optional int64 blksize = 10 [default = 0];
    optional int64 blocks = 11 [default = 0];
    optional int64 atime = 12 [default = 0];
    optional int64 mtime = 13 [default = 0];
    optional int64 ctime = 14 [default = 0];
}

message CreateFile {
    required string file_id = 1;
    required uint32 mode = 2;
}

message DeleteFileAtStorage {
    required string file_id = 1;
}

message TruncateFile {
    required string file_id = 1;
    required int64 length = 2;
}

message ReadFile {
    required string file_id = 1;
    required int64 size = 2;
    optional int64 offset = 3 [default = 0];
}

message FileData {
    required string answer_status = 1;
    optional string operation_guid = 2;
    optional string new_data_source = 3;
    optional bytes data = 4;
}

message WriteFile {
    required string file_id = 1;
    optional int64 size = 2;
    optional bytes data = 3;
    optional int64 offset = 4 [default = 0];
}

message WriteInfo {
    required string answer_status = 1;
    optional int64 bytes_written = 2 [default = 0];
    optional string operation_guid = 3;
    optional string new_data_listener = 4;
}

message ChangePermsAtStorage {
    required string file_id = 1;
    required int32 perms = 2;
}

